compiler/src/parsing/ast_helper.re

```diff
@@ -85,6 +100,21 @@ module CDecl = {
   };
   let singleton = (~loc=?, n) => mk(~loc?, n, PConstrSingleton);
   let tuple = (~loc=?, n, a) => mk(~loc?, n, PConstrTuple(a));
+  let record = (~loc=?, n, a) => {
+    List.iter(
+      ld =>
+        if (ld.pld_mutable == Mutable) {
+          raise(
+            SyntaxError(
+              ld.pld_loc,
+              "An inline record constructor cannot have mutable fields.",
```

Comment:
This specific issue was never really discussed but since right now we are only allowing the record fields to be accessed by patterns matched to the fields, mutable fields would be pointless (e.g. `x.field = ...` is not possible currently)

Reply:
Oh, I guess I was thinking `Rec{ foo }` would allow me to mutate `foo` but I suppose that's an immutable destructure, so it makes sense that mutable values wouldn't be allowed.

How crazy would it be to support `Rec{ mut foo }` and then changing `foo` would be actually change the inline record? Just thinking for future improvements

Reply:
This would be a very involved change, but I was hoping to use this syntax in 0.6 for mutable patterns, e.g. `let (mut fst, mut snd) = ...` instead of `let mut (fst, snd) = ...`.

This just makes the binding mutable though, it wouldn't change the constructor itself.

(Just throwing in my fly-by comments.)